openverse_catalog/docs/provider_data_ingester_faq.md

```diff
@@ -0,0 +1,142 @@
+# ProviderDataIngester FAQ
+
+The most straightforward implementation of a `ProviderDataIngester` repeats the following process:
+
+* Builds a set of query params for the next request, based upon the previous params (for example, by updating offsets or page numbers)
+* Makes a single GET request to the configured static `endpoint` using the built params
+* Extracts a "batch" of records from the response, as a list of record JSON representations
+* Iterates over the records in the batch, extracting desired data, and commits them to local storage
+
+Some provider APIs may not fit neatly into this workflow. This document addresses some common use cases.
+
+## How do I process a provider "record" that contains data about multiple records?
+
+**Example**: You're pulling data from a Museum database, and each "record" in a batch contains multiple photos of a single physical object.
+
+**Solution**: The `get_record_data` method takes a `data` object representing a single record from the provider API. Typically, it extracts required data and returns it as a single dict. However, it can also return a **list of dictionaries** for cases like the one described, where multiple Openverse records can be extracted.
+
+```python
+def get_record_data(self, data: dict) -> dict | list[dict] | None:
+    records = []
+
+    for record in data.get("associatedImages", []):
+        # Perform data processing for each record and add it to a list
+        records.append({
+            "foreign_landing_url": record.get("foreign_landing_url")
+            ...
+        })
+
+    return records
+```
+
+## What if I can't get all the necessary information for a record from a single API request?
+
+**Example**: A provider's `search` endpoint returns a list of image records containing *most* of the information we need for each record, but not image dimensions. This information *is* available via the API by hitting a `details` endpoint for a given image, though.
+
+**Solution**: In this case, you can reuse the `get_response_json` method by passing in the endpoint you need:
+
+```python
+def get_record_data(self, data: dict) -> dict | list[dict] | None:
+    ...
+
+    # Get data from the details endpoint
+    response_json = self.get_response_json(
+        query_params={"my_params": "foo"},
+        endpoint=f"https://foobar.museum.org/api/v1/images/{data.get("uuid")}"
+    )
+
+    ...
+```
+
+When doing this, keep in mind that adding too many requests may slow down ingestion. Be aware of rate limits from your provider API as well.
+
+## What if my API endpoint isn't static and needs to change from one request to another?
+
+Example: Rather than passing a `page` number in query parameters, a provider expects the `page` as part of the endpoint path itself.
```

Comment:
```suggestion
**Example**: Rather than passing a `page` number in query parameters, a provider expects the `page` as part of the endpoint path itself.
```